rust/kernel/src/filesystem.rs

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: GPL-2.0
```

Comment:
This module is really incomplete, in terms of things that are a good fit for upstream, I think we might want to delete it.

Reply:
Yeah, I kept the files as-is as much as possible to make the first push easy and let you diff easily (I could have made a small commit with only your files and then this one, though...). Later we can take out things (i.e. just to clarify: we are not going to send this right -- we will iterate here first for some weeks/months), so feel free to do so!

Reply:
Added comment.